File: backend/models/transaction.js
const mongoose = require('mongoose');

const transactionSchema = new mongoose.Schema({
    description: {
        type: String,
        required: true
    },
    amount: {
        type: Number,
        required: true
    },
    date: {
        type: String,
        required: true
    }
});

module.exports = mongoose.model('Transaction', transactionSchema);

File: backend-java/src/main/java/com/ninfinances/domain/model/Transaction.java
package com.ninfinances.domain.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@Document(collection = "transactions")
public class Transaction {
    @Id
    private String id;
    private String description;
    private Double amount;
    private String date;
    private String userId;
} 